Integrated Zephyr shell with the CHIP lib shell (#39)
kkasperczyk-no/connectedhomeip: docs/guides/nrfconnect_examples_cli.md

They are currently used for testing purposes and allow only to get some
information about CHIP stack state, but not to modify it.

To enable using CHIP commands, set the `CONFIG_CHIP_LIB_SHELL=y` Kconfig
option in the `prj.conf` file of the given example.

Every invoked command must be preceded by the `matter` prefix.

See the following subsections for the description of each CHIP-specific command.

> ### `onboardingcodes`

Reviewer: Remove the ticks, given the edits to other command headers?

> Handles group of commands used to view information about device onboarding codes. You can use it without any subcommand to print all available onboarding codes or add a specific subcommand.

Reviewer: ```suggestion
Handles a group of commands that are used to view information about device onboarding codes. You can use this command without any subcommand to print all available onboarding codes or to add a specific subcommand.
```

```shell
uart:~$ matter onboardingcodes
QRCode:             MT:W0GU2OTB00KA0648G00
QRCodeUrl:          https://dhrishi.github.io/connectedhomeip/qrcode.html?data=MT%3AW0GU2OTB00KA0648G00
ManualPairingCode:  34970112332
```

> It can also take the following subcommands presented below.

Reviewer: ```suggestion
The `onboardingcodes` command can also take the subcommands listed below.
```

> #### qrcode

Reviewer: I think the current hierarchy works well, but indeed we could include the name of the main command in the subcommand header, like `#### (onboardingcodes) qrcode`

Also, shouldn't this `####` for qrcode be a `###`?

> #### qrcode

Author: If it comes to the including command names in subcommands - I don't find it useful, as it's a lack of consequence, because following this pattern we should also add `matter` command. That will result in having listed commands like `matter ble adv start`, what IMO is not looking good, especially that we have examples of usage for each command.

if it comes to using ### instead of ### I did it intentionally to highlight main group command from the subcommands. Is it wrong and all should be on the same level?

Prints the device
[onboarding QR code payload](https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/nrfconnect_android_commissioning.md#preparing-accessory-device).
Takes no arguments.

```shell
uart:~$ matter onboardingcodes qrcode
MT:W0GU2OTB00KA0648G00
```

#### qrcodeurl

Prints the URL to view the
[device onboarding QR code](https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/nrfconnect_android_commissioning.md#preparing-accessory-device)
in a web browser. Takes no arguments.

```shell
uart:~$ matter onboardingcodes qrcodeurl
https://dhrishi.github.io/connectedhomeip/qrcode.html?data=MT%3AW0GU2OTB00KA0648G00
```

#### manualpairingcode

Prints the pairing code for the manual onboarding of a device. Takes no
arguments.

```shell
uart:~$ matter onboardingcodes manualpairingcode
34970112332
```

> ### `config`

Reviewer: Ticks again, not sure if needed given other edits.

> Handles group of commands used to view device configuration information. You can use it without any subcommand to print all available configuration data or add a specific subcommand.

Reviewer: ```suggestion
Handles a group of commands that are used to view device configuration information. You can use this command without any subcommand to print all available configuration data or to add a specific subcommand.
```

```shell
VendorId:        9050 (0x235A)
ProductId:       20043 (0x4E4B)
ProductRevision: 1 (0x1)
FabricId:        
PinCode:         020202021
Discriminator:   f00
DeviceId:        
```

> It can also take the following subcommands presented below.

Reviewer: ```suggestion
The `config` command can also take the subcommands listed below.
```

#### pincode

Prints the PIN code for device setup. Takes no arguments.

```shell
uart:~$ matter config pincode
020202021
```

#### discriminator

Prints the device setup discriminator. Takes no arguments.

```shell
uart:~$ matter config discriminator
f00
```

#### vendorid

Prints the vendor ID of the device. Takes no arguments.

```shell
uart:~$ matter config vendorid
9050 (0x235A)
```

#### productid

Prints the product ID of the device. Takes no arguments.

```shell
uart:~$ matter config productid
20043 (0x4E4B)
```

#### productrev

Prints the product revision of the device. Takes no arguments.

```shell
uart:~$ matter config productrev
1 (0x1)
```

#### deviceid

Prints the device identifier. Takes no arguments.

#### fabricid

Prints the fabric identifier. Takes no arguments.

> ### `ble`

Reviewer: Ticks again, not sure if needed.

> Handles group of commands used to control device Bluetooth LE transport state. You need to use it along with ont of the additional subcommands listed below.

Reviewer: ```suggestion
Handles a group of commands that are used to control the device Bluetooth LE transport state. You must use this command together with one of the additional subcommands listed below.
```

#### help

Prints help information about `ble` commands group.

```shell
uart:~$ matter ble help
  help            Usage: ble <subcommand>
  adv             Enable or disable advertisement. Usage: ble adv <start|stop|state>
```

#### adv start

Enables Bluetooth LE advertising.

```shell
uart:~$ matter ble adv start
Starting BLE advertising
```

#### adv stop

Disables Bluetooth LE advertising.

```shell
uart:~$ matter ble adv stop
Stopping BLE advertising
```

#### adv status

Prints the information about the current Bluetooth LE advertising status.

```shell
uart:~$ matter ble adv state
BLE advertising is disabled

```

> ### `nfc`

Reviewer: Tick.

> Handles group of commands used to control device NFC tage emulation state. You need to use it along with ont of the additional subcommands listed below.

Reviewer: ```suggestion
Handles a group of commands that are used to control the device NFC tag emulation state. You must use this command togethew with one of the additional subcommands listed below.
```

> Handles a group of commands that are used to control the device NFC tag emulation state. You must use this command togethew with one of the additional subcommands listed below.

Reviewer 2: togethe**r**
